index.d.ts

```diff
@@ -0,0 +1,80 @@
+// Type definitions for next-mdx-remote
+// Project: https://github.com/hashicorp/next-mdx-remote
+// Definitions by: Steven Schmatz <https://github.com/stevenschmatz>
+
+declare module 'next-mdx-remote/render-to-string' {
+  import * as React from 'react';
+
+  /**
+   * Runs the MDX renderer on the MDX string provided with the components and data provided.
+   */
+  export default function renderToString(
+    /** Raw MDX contents as a string. */
+    source: string,
+    /** Optional parameters, such as components, plugins, and data. */
+    params?: {
+      /**
+       * A object mapping names to React components.
+       * The key used will be the name accessible to MDX.
+       *
+       * For example: `{ ComponentName: Component }` will be accessible in the MDX as `<ComponentName/>`.
+       */
+      components?: { 
+        [componentName: string]: React.FunctionComponent | React.Component;
```

Comment:
When I was working on type definitions, I also found that I needed to specify this type in this way instead of just `React.Element`, but was confused about why. Do you have any further insight on this one?